src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.service.ts

```diff
@@ -85,6 +84,13 @@ export class IdeSettingsService {
      */
     public deleteIdePreference(programmingLanguage: ProgrammingLanguage): Observable<HttpResponse<void>> {
         const params = new HttpParams().set('programmingLanguage', programmingLanguage);
-        return this.http.delete<void>(this.ideSettingsUrl, { params, observe: 'response' });
+        return this.http.delete<void>(this.ideSettingsUrl, { params, observe: 'response' }).pipe(
+            map((response) => {
+                const prev = new Map(this.idePreferencesSubject.getValue());
+                prev.delete(programmingLanguage);
+                this.idePreferencesSubject.next(prev);
+                return response;
```

Comment:
Same here, you can also just use `tap`, even without the response parameter, since you don't interact with it here